esx_service/vmodl/TenantManager.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2016 VMware, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Copyright 2016 VMware, Inc.  All rights reserved. 
+Licensed under the Apache License, Version 2.0 
+http://www.apache.org/licenses/LICENSE-2.0
+"""
+
+from VmodlDecorators import ManagedType, EnumType, Method, \
+   Return, RegisterVmodlTypes, F_OPTIONAL, Param, DataType, Attribute
+from pyVmomi import Vmodl
+from pyVmomi.VmomiSupport import newestVersions
+
+try:
+   from asyncVmodlEmitterLib import JavaDocs, Internal
+except ImportError:
+   pass
+   def JavaDocs(parent, docs):
+      def Decorate(f):
+         return f
+      return Decorate
+   def Internal(parent):
+      def Decorate(f):
+         return f
+      return Decorate
+
+
+# _VERSION = newestVersions.Get("vim")
+_VERSION = 'vim.version.version10'
+
+# Vmodl Names
+
+class TenantManager:
+   ''' This class manages the lifecycle of tenants for vSphere Container Service '''
+
+   _name = "vim.vcs.TenantManager"
+
+   @Internal(parent=_name)
+   @ManagedType(name=_name, version=_VERSION)
+   def __init__(self):
+       pass
+
+   @JavaDocs(parent=_name, docs=
+   """
+   Create a Tenant instance
+   """
+   )
+   @Method(parent=_name, wsdlName="CreateTenant")
+   @Param(name="name", typ="string")
+   @Param(name="description", typ="string", flags=F_OPTIONAL)
+   @Return(typ="vim.vcs.Tenant")
+   def CreateTenant(self, name, description=None):
+       pass
+
+   @JavaDocs(parent=_name, docs=
+   """
+   Remove a Tenant instance
+   """
+   )
+   @Method(parent=_name, wsdlName="RemoveTenant")
+   @Param(name="name", typ="string")
```

Comment:
Since all Tenant related APIs use tenant name as the unique key, I think it makes sense to also use name as the key for this API. What do you think?